crates/net/network/src/session/active.rs

```diff
@@ -641,9 +641,11 @@ impl Future for ActiveSession {
         while this.internal_request_timeout_interval.poll_tick(cx).is_ready() {
             // check for timed out requests
             if this.check_timed_out_requests(Instant::now()) {
-                let _ = this.to_session_manager.clone().try_send(
-                    ActiveSessionMessage::ProtocolBreach { peer_id: this.remote_peer_id },
-                );
+                if let Poll::Ready(Ok(_)) = to_session_manager_poll_tx.poll_reserve(cx) {
+                    let _ = to_session_manager_poll_tx.send_item(
+                        ActiveSessionMessage::ProtocolBreach { peer_id: this.remote_peer_id },
+                    );
```

Comment:
This does not quite have the same effect as before,

here I think we can just set the message to `pending_message_to_session`, which will make sure the message gets delivered

Reply:
Is this what you mean?
```rust
                if this.check_timed_out_requests(Instant::now()) {
                    self.pending_message_to_session = ActiveSessionMessage::ProtocolBreach { peer_id: this.remote_peer_id };
                }
```